### PR TITLE
20201202

### PR DESCRIPTION
**Pull request by Yolanda18-star**
@langyuemeng @YiniXu9506 @YangKeao PTAL

### CODE REVIEW

File: ture-technology-in-chaos-mesh.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+2. 使用 Mutating Webhook 修改 Pod 的启动命令，比如本身启动命令是 `/app`，我们要将它修改成 `/waitfs.sh /app`，而我们提供的 `waitfs.sh` 会不断检查文件系统是否已经挂载成功，如果已经成功就再启动 `/app` 。
+
+3. 自然的，我们依旧使用 Mutating Webhook 来在 Pod 中多加入一个容器用于运行 ChaosFS。运行 ChaosFS 的容器需要与目标容器共享某个 volume，比如 `/mnt`。然后将它挂载至目标目录，比如 `/mnt/a`。同时开启适当的 mount propagation ，来让 ChaosFS 容器的 volume 中的挂载穿透（share）至 host，再由 host 穿透（slave）至目标。（如果你了解 mnt namespace 和 mount ，那么一定知道 share 和 slave 是什么意思）。
+
+这样一来，就完成了对目标程序 IO 过程的注入。但它是如此的不好用：
+
+1. 只能对某个 volume 的子目录注入，而无法对整个 volume 注入。
+
+2. 要求 Pod 中明文写有 command，而不能是隐含使用镜像的 command 。因为如果使用镜像隐含的 command 的话，/waitfs.sh` 就不知道在挂载成功之后应该如何启动应用了。
```
Comment (YangKeao) [inline ture-technology-in-chaos-mesh.md:67]: /waitfs.sh\`  => \`/waitfs.sh\`

File: ture-technology-in-chaos-mesh.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
+---
```
Comment (YangKeao) [inline ture-technology-in-chaos-mesh.md:1]: `Chaos Mesh 技术内幕` will be a series of articles and there will be a lot of articles starting with `Chaos Mesh 技术内幕 | `. Does it matter to name it `ture-technology-in-chaos-mesh.md`?

A full name could be `chaos-mesh-internals-how-to-inject-io-faults-during-runtime`. I don't know whether it's too long.

**Review by YangKeao — COMMENTED** [review]
Rest LGTM

**Review by YangKeao — APPROVED** [review]
LGTM

**Comment by langyuemeng** [comment]
LGTM

**Review by YiniXu9506 — APPROVED** [review]
LGTM

**Pull request merged**